manifests/package.pp

```diff
@@ -10,11 +10,11 @@
   }
 
   $caddy_url        = 'https://caddyserver.com/download/linux'
-  $caddy_dl_url     = "${caddy_url}/${caddy::params::arch}?plugins=${caddy::caddy_features}&license=${caddy::caddy_license}&telemetry=${caddy::caddy_telemetry}"
-  $caddy_dl_dir     = "${caddy::params::caddy_tmp_dir}/caddy_linux_${$caddy::params::arch}_custom.tar.gz"
-  $caddy_dl_command = $caddy::params::caddy_license ? {
+  $caddy_dl_url     = "${caddy_url}/${caddy::arch}?plugins=${caddy::caddy_features}&license=${caddy::caddy_license}&telemetry=${caddy::caddy_telemetry}"
```

Comment:
Would be great to see this other pattern of with `::package` and  `::config` also removed and all just condensed into the main class.